sql/core/src/main/scala/org/apache/spark/sql/execution/python/BatchEvalPythonExec.scala

```diff
@@ -98,7 +353,7 @@ case class BatchEvalPythonExec(udfs: Seq[PythonUDF], output: Seq[Attribute], chi
       // For each row, add it to the queue.
       val inputIterator = iter.grouped(100).map { inputRows =>
         val toBePickled = inputRows.map { inputRow =>
-          queue.add(inputRow)
+          queue.add(inputRow.asInstanceOf[UnsafeRow])
```

Comment:
Comparing to assert, the error message actually looks similar: the row should be UnsafeRow but actually it's not. I think this is an fundamental assumption, there are many places (20 for now) that has this.
```
davies@localhost:~/work/spark$ git grep '.asInstanceOf\[UnsafeRow\]' | wc -l
20
```